### PR TITLE
LPS-181462 | Delete API Application

### PR DESCRIPTION
**Pull request by Hijky**
Background:
Add a testcase to delete API Application

Test Plan
CanSeeDeleteApiApplicationDialog
Given create an API application in API Builder
When from dropdown menu I click Delete option
Then I can see the prompt message in the Delete API Application dialog

CannotDeleteApiWithIncorrectTitleValue
Given create an API application in API Builder with title ‘test'
And Given from dropdown menu I click Delete option
When I enter ‘Test’ as the API title
Then I can see message about type the API title
And Then Delete button is disabled

CanDeleteAPIFromApiBuilderAndApi
Given create an API application in API Builder
And Given from dropdown menu I click Delete option
When I confirm deletion by typing the API title
Then the API is not present in API Builder
And Then the API is removed from API

CanDeleteUnpublishedApiAndAssociatedEntities
Given create an API application in API Builder
And Given with postAPISchema to create a Schema associate to the API application
And Given with postAPIEndpoint to create an Endpoint associate to API application and Schema
When I delete the API application
Then the API is not present in API Builder
And Then the associated Endpoint and Schema are deleted

CanDeletePublishedApiAndAssociatedEntities
Given publish an API application in API Builder
And Given with postAPISchema to create a Schema associate to the API application
And Given with postAPIEndpoint to create an Endpoint associate to API application and Schema
When I delete the API application
Then the API is not present in API Builder
And Then the associated Endpoint and Schema are deleted

Jira ticket:
https://liferay.atlassian.net/browse/LPS-181462

### CODE REVIEW

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/SchemaAPI.macro [hunk 1]
```diff
@@ -58,7 +58,7 @@ definition {
 	}
 
 	macro getAPISchemas {
-		var curl = ApplicationAPI._curlAPIApplication(virtualHost = ${virtualHost});
+		var curl = SchemaAPI._curlAPISchema(virtualHost = ${virtualHost});
```
Comment (yanan) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/SchemaAPI.macro:61]: :+1:

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/APIBuilderUI.macro [hunk 2]
```diff
@@ -60,6 +60,19 @@ definition {
 			locator1 = "Button#ANY");
 	}
 
+	macro goToDeleteAPIApplication {
```
Comment (yanan) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/APIBuilderUI.macro:63]: Maybe it's better to name it like deleteAPIApplication?

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/paths/APIBuilder.path [hunk 3]
```diff
@@ -105,6 +110,16 @@
 	<td>//input[(@placeholder='Enter title.')]</td>
 	<td></td>
 </tr>
+<tr>
```
Comment (yanan) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/paths/APIBuilder.path:113]: I've tried locally that seems the MODAL_DELETE_BUTTON and DELETE_BUTTON_DISABLED are the same Delete button in modal, could you use one?

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/paths/APIBuilder.path [hunk 4]
```diff
@@ -105,6 +110,16 @@
...
+	<td>MODAL_DELETE_BUTTON</td>
+	<td>//button[@id='modalDeleteButton']</td>
+	<td></td>
+</tr>
+<tr>
```
Comment (yanan) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/paths/APIBuilder.path:118]: How about naming this something like TYPE_TITLE_IN_DELETE_MODAL?

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/DeleteAPIApplication.testcase [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+			PortalInstances.tearDownCP();
+		}
+	}
+
+	@priority = 4
+	test CanDeleteAPIFromApiBuilderAndApi {
+		property portal.acceptance = "true";
+		property test.liferay.virtual.instance = "false";
+
+		task ("And Given from dropdown menu I click Edit option") {
```
Comment (yanan) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/DeleteAPIApplication.testcase:40]: We should copy the description in task("") from the correct ticket, I think it should be "click Delete" here, please take a look and update, thanks

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/DeleteAPIApplication.testcase [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+			var totalCount = JSONUtil.getWithJSONPath(${response}, "$..totalCount");
+
+			TestUtils.assertEquals(
+				actual = ${totalCount},
+				expected = 0);
+		}
+	}
+
+	@priority = 4
+	test CanDeletePublishedApiAndAssociatedEntities {
```
Comment (yanan) [inline modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/DeleteAPIApplication.testcase:70]: Just wondering maybe we can move CanDeletePublishedApiAndAssociatedEntities and the following CanDeleteUnpublishedApiAndAssociatedEntities into interation test? @magjed4289

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/paths/APIBuilder.path (hunk 3 above)
Comment (Hijky) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/paths/APIBuilder.path:113]: I don't think so, In the testcase I need to test the button status is disabled

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/APIBuilderUI.macro (hunk 2 above)
Comment (Hijky) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/APIBuilderUI.macro:63]: deleteAPIApplication we has the same name in ApplicationAPI.macro

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/paths/APIBuilder.path (hunk 3 above)
Comment (Hijky) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/paths/APIBuilder.path:113]: I found we can use button#ANY_DISABLED to replace the DELETE_BUTTON_DISABLED

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/DeleteAPIApplication.testcase (hunk 6 above)
Comment (magjed4289) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/DeleteAPIApplication.testcase:70]: yes if it is only API layer test we can move it to the integration: analysing first whether a similar test is already there.

Comment (magjed4289) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/DeleteAPIApplication.testcase:70]: but this story is about the UI so I don't understand why the deletion is being made through API

**Review by magjed4289 — CHANGES_REQUESTED** [review]

File: modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/DeleteAPIApplication.testcase (hunk 6 above)
Comment (yanan) [reply modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/tests/[UI]AllowUsersToDefineAPIApplicationsInTheAPIBuilder/DeleteAPIApplication.testcase:70]: good point, yes, we should update to delete by UI, thanks!

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:stop

**Comment by Hijky** [comment]
ci:test:sf

**Comment by Hijky** [comment]
ci:test:headless-functional:nocompile

**Comment by Hijky** [comment]
The failures are not related to PR @magjed4289 @yanan

**Review by magjed4289 — APPROVED** [review]

**Comment by magjed4289** [comment]
loads of tests were aborted so I'm running the suite again. just in case

**Comment by magjed4289** [comment]
ci:test:headless-functional:nocompile

**Comment by Hijky** [comment]
The failures are two test-fix https://github.com/liferay-headless/liferay-portal/pull/1434 https://github.com/liferay-headless/liferay-portal/pull/1437, I think we can forward this PR
@magjed4289 @yanan

**Comment by Hijky** [comment]
ci:forward:force

**Pull request closed without merging**